docs/proposal_pytorch_distribution.md

```diff
@@ -0,0 +1,335 @@
+   aware of channels and dependencies, so it should work with very little
+   extra effort.
+
+
+### What happens when a new PyTorch release is made?
+
+For minor or major versions of PyTorch, new releases of downstream packages
+will also be necessary. A number of packages, such as `torchvision`,
+`torchaudio` and `torchtext`, are anyway released in sync. Other packages in
+the `pytorch` channel may need to be manually released via a PR to the
+`conda-distro` repo).
+
+Version constraints should be set such that a bugfix release of PyTorch does
+not require any new downstream package releases.
+
+
+### Dealing with packages that aren't maintained
+
+Proposing a package for inclusion in the `pytorch` channel implies a
+commitment to keep maintaining the package. There wil be a place to list one
+or more maintainers for each package so they can be pinged if needed. In case
+a package is not up-to-date or broken and it does not get fixed, after a
+certain duration (length TBD) it may be removed from the channel.
+
+
+## Alternatives
+
+### Conda-forge
+
+The main alternative to making the `pytorch` channel an integration channel
+that distributes many packages that depend on PyTorch is to have a
+(GPU-enabled) PyTorch package in conda-forge, and tell users and package
+authors that that is the place to go. It will require working with
+conda-forge in order to ensure that the `pytorch` package is of high quality,
+either by copying over the binaries from the `pytorch` channel or by
+migrating recipes and keeping them in sync. See
+[this very long discussion](https://github.com/conda-forge/pytorch-cpu-feedstock/issues/7)
+for details (and issues).
+
+Advantages of this alternative are:
+
+- Conda-forge already has a lot of packages, and there are established tools
+  and processes for adding and updating them. Which means it's less likely
+  for there to be issues with dependencies (e.g. packages with many or
+  unusual dependencies may not be accepted into the `pytorch` channel, while
+  `conda-forge` will be fine with them).
+- Users are likely already familiar with using the `conda-forge` channel.
+
+Disadvantages of this alternative are:
+
+- As of today, conda-forge doesn't have GPU hardware. Building is stil
+  possible using CUDA stubs, however testing cannot really be done inside CI,
+  only manually (which is a pain, especially when having to test multiple
+  hardware and OS platforms).
+- Conda-forge uses a single compiler toolchain for all packages it builds for
+  a given platform. Currently that is known to work with PyTorch on Linux,
+  and unknown (but likely works) on other platforms. That's not guaranteed to
+  always remain the case though; conda-forge compilers are typically fairly
+  old (to stay roughly in sync with `defaults`, which considers HPC use cases
+  with older CentOS releases), while PyTorch may have the need to start
+  requiring a newer compiler at some point. This would be a serious problem.
```

Comment:
I'll add another disadvantage: release coordination. For a PyTorch release, packages for pytorch, torchvision, torchtext, torchaudio will all be built together and then released. Building a set of packages like that depend on each other and releasing them in a coordinated fashion is hard to do on conda-forge, given that if everything is in feedstocks, the new pytorch package must already be available before the next build can start. And then you get delays on public CI etc - difficult to do in a single day.